cantusdb/turtle/csv2rdf_single_subject.py

```diff
@@ -0,0 +1,74 @@
+import csv
+import validators
+import sys
+import json
+import os
+from rdflib import Graph, URIRef, Literal
+from rdflib.namespace import RDF
+from typing import Optional
+
+dirname = os.path.dirname(__file__)
+mappername = os.path.join(dirname, 'relations_mapping.json')
+destname = os.path.join(dirname, 'out_rdf.ttl')
+
+def convert_csv_to_turtle(filename) -> Graph:
+    """
+    (str) -> Graph
+
+    Adds all informations as RDF triples from the input filename into a graph and return it.
+    *Important: This input file must have the first column as subjects of all triples
+    
+    @Pre: type(filename) == str
+    """
+    with open(filename, "r", encoding="utf-8") as csv_file:
+        # TODO: maybe change this one to accomadate for multiple files
+
+        g = Graph()
+
+        csv_reader = csv.reader(csv_file)
+        ontology_dict = json.load(open(mappername, "r"))
+
+        header = next(csv_reader)
+        header_without_subject = header[1:]
+        predicates = []
+        for column in header_without_subject:
+            if column in ontology_dict:
+                predicates.append(URIRef(ontology_dict[column]))
+
+        ontology_type: Optional[str] = ontology_dict["type"]
```

Comment:
This looks like another requirement: the `ontology_dict` must contain a key of `type`. Please document this.

Reply:
No, it does not have to contain "type". My previous implementation is: if there is no "type", then it is None.

Reply:
If there is no type key, this code will raise a KeyError Exception. You want: 

```
 ontology_type: Optional[str] = ontology_dict.get("type")
```

That will return `None` if it does not have a type key.

Reply:
Ok, @Yueqiao12Zhang 

If you fix this and if you provide a clearer description in the PR text (at the top of this page) and/or in a linked issue, I will approve.